exec/java-exec/src/main/java/org/apache/drill/exec/planner/ParquetPartitionDescriptor.java

```diff
@@ -125,4 +117,16 @@ private String getBaseTableLocation() {
     final FormatSelection origSelection = (FormatSelection) scanRel.getDrillTable().getSelection();
     return origSelection.getSelection().selectionRoot;
   }
+
+  @Override
+  protected void createPartitionSublists() {
+    Set<String> fileLocations = ((ParquetGroupScan) scanRel.getGroupScan()).getFileSet();
+    List<PartitionLocation> locations = new LinkedList<>();
+    for (String file: fileLocations) {
+      locations.add(new DFSPartitionLocation(MAX_NESTED_SUBDIRS, getBaseTableLocation(), file));
```

Comment:
Is fileLocation the absolute path to the file, or the relative path to the selectionRoot? If it's the former case, the file name could be quite long, considering that the directory could locate deep in the file system tree.  For instance, DRILL-1488 tried to consider the case where the file identifier could be as long as 1024, the default identifier max length.  So, in such extreme cases, we may hit heap memory issue, if we keep 100k long absolute path in the heap. Certainly, I agree it's not the normal cases, though it could happen.